contracts/core/OperationExecutor.sol

```diff
@@ -153,4 +160,43 @@ contract OperationExecutor is IERC3156FlashBorrower {
 
     return keccak256("ERC3156FlashBorrower.onFlashLoan");
   }
+
+  function receiveFlashLoan(
+    IERC20[] memory tokens,
+    uint256[] memory amounts,
+    uint256[] memory feeAmounts,
+    bytes memory data
+  ) external override {
+
+    address lender = registry.getRegisteredService(BALANCER_VAULT);
+    require(msg.sender == lender, "Untrusted flashloan lender");
+
+    BalancerFlashloanData memory flData = abi.decode(data, (BalancerFlashloanData));
+    address initiator = flData.borrower;
+    IERC20 borrowedToken = tokens[0];
+
+    require(borrowedToken.balanceOf(address(this)) >= flData.amount, "Flashloan inconsistency");
+
+    if (flData.dsProxyFlashloan) {
+      borrowedToken.safeTransfer(initiator, flData.amount);
```

Comment:
Could pull this logic out into a private method and share between flashloan cb handlers? Just because OpExecutor is probably getting fairly chunky now.